mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -45,45 +64,53 @@ func runIgnitionFailure(c cluster.TestCluster) {
 	}
 }
 
-func ignitionFailure(c cluster.TestCluster) error {
-	// We can't create files in / due to the immutable bit OSTree creates, so
-	// this is a convenient way to test Ignition failure.
-	failConfig, err := conf.EmptyIgnition().Render(conf.FailWarnings)
-	if err != nil {
-		return errors.Wrapf(err, "creating empty config")
+func runDualBootfsFailure(c cluster.TestCluster) {
+	if err := dualBootfsFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	failConfig.AddFile("/notwritable.txt", "Hello world", 0644)
+}
 
-	builder := platform.NewQemuBuilder()
-	defer builder.Close()
-	builder.SetConfig(failConfig)
-	err = builder.AddBootDisk(&platform.Disk{
-		BackingFile: kola.QEMUOptions.DiskImage,
-	})
-	if err != nil {
-		return err
+func runDualBootfsIgnitionFailure(c cluster.TestCluster) {
+	if err := dualBootfsIgnitionFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	builder.MemoryMiB = 1024
-	builder.Firmware = kola.QEMUOptions.Firmware
+}
+
+// take string and grep for it in console logs
+func VerifyError(builder *platform.QemuBuilder, tempLogFile string, searchString string) error {
 	inst, err := builder.Exec()
 	if err != nil {
 		return err
 	}
 	defer inst.Destroy()
-
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+
 	defer cancel()
 
 	errchan := make(chan error)
 	go func() {
-		err := inst.WaitAll(ctx)
-		if err == nil {
-			err = fmt.Errorf("Ignition unexpectedly succeeded")
-		} else if err == platform.ErrInitramfsEmergency {
-			// The expected case
-			err = nil
+		resultingError := inst.WaitAll(ctx)
+		if resultingError == nil {
+			resultingError = fmt.Errorf("ignition unexpectedly succeeded")
+		} else if resultingError == platform.ErrInitramfsEmergency {
+			// Expectred initramfs failure, checking the console file to insure
+			// that coreos.ignition.failure failed
+			b, err := os.ReadFile(tempLogFile)
+			if err != nil {
+				resultingError = err
+			}
+			textExists, err := regexp.Match(searchString, b)
+			if err != nil {
+				resultingError = err
+			}
+			if textExists {
+				// The expected case
+				resultingError = nil
+			} else {
+				resultingError = errors.Wrapf(err, "expected coreos.ignition.failure to fail")
+			}
 		} else {
-			err = errors.Wrapf(err, "expected initramfs emergency.target error")
+			resultingError = errors.Wrapf(err, "expected initramfs emergency.target error")
 		}
 		errchan <- err
```

Comment:
There's something wrong with error-handling above. Did you mean for this to be `errchan <- resultingError`?